conans/client/toolchain/cmake/base.py

```diff
@@ -14,24 +38,38 @@ class CMakeToolchainBase(object):
 
     def __init__(self, conanfile, *args, **kwargs):
         self._conanfile = conanfile
+        self.variables = Variables()
+        self.preprocessor_definitions = Variables()
+        try:
+            # This is only defined in the cache, not in the local flow
+            self.install_prefix = self._conanfile.package_folder.replace("\\", "/")
+        except AttributeError:
+            # FIXME: In the local flow, we don't know the package_folder
+            self.install_prefix = None
 
     def _get_template_context_data(self):
         """ Returns two dictionaries, the context for the '_template_toolchain' and
             the context for the '_template_project_include' templates.
         """
-        return {}, {}
+        ctxt_toolchain = {
+            "variables": self.variables,
+            "variables_config": self.variables.configuration_types,
+            "preprocessor_definitions": self.preprocessor_definitions,
+            "preprocessor_definitions_config": self.preprocessor_definitions.configuration_types,
+            "install_prefix": self.install_prefix,
```

Comment:
We probably want to discuss a bit about the UI of this when this is local dev flow